src/Analyzers/Core/CodeFixes/RemoveRedundantEquality/RemoveRedundantEqualityCodeFixProvider.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.CodeAnalysis.RemoveRedundantEquality
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic), Shared]
+    internal sealed class RemoveRedundantEqualityCodeFixProvider
+        : SyntaxEditorBasedCodeFixProvider
+    {
+        [ImportingConstructor]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+        public RemoveRedundantEqualityCodeFixProvider()
+        {
+        }
+
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(IDEDiagnosticIds.RemoveRedundantEqualityDiagnosticId);
+
+        internal override CodeFixCategory CodeFixCategory => CodeFixCategory.CodeStyle;
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                context.RegisterCodeFix(new MyCodeAction(
+                    AnalyzersResources.Remove_redundant_equality,
+                    c => FixAsync(context.Document, diagnostic, c)),
+                    diagnostic);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        protected override async Task FixAllAsync(Document document, ImmutableArray<Diagnostic> diagnostics, SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            foreach (var diagnostic in diagnostics)
+            {
+                var node = root.FindNode(diagnostic.AdditionalLocations[0].SourceSpan, getInnermostNodeForTie: true);
+                var replacementNode = root.FindNode(diagnostic.AdditionalLocations[1].SourceSpan, getInnermostNodeForTie: true).NormalizeWhitespace();
```

Comment:
definitely do not call NormalizeWhitespace.  that will destroy any specific user formatting someone may have in their code. indeed, we should now test that that doesn't happen.

Reply:
@CyrusNajmabadi That's what came to mind to remove the extra space. Is there a way to just remove the trailing whitespace from the node without affecting other whitespaces?

Reply:
I would just add an elastic whitespace around the node, indicating to the formatter that it can fixup whitespace here.

Reply:
@CyrusNajmabadi Do you mean to just use `.WithTrailingTrivia(elasticSpace)` (where elasticSpace is a language-specific SyntaxTrivia passed from subclasses)? or get the current trivia and add elasticSpace to them?

Reply:
I would add SyntaxFactory.ElasticTrivia to the end of the existing trivia

Reply:
@CyrusNajmabadi I can't find any field/property named "ElasticTrivia". There is an `ElasticSpace` but they're defined in the language-specific classes. So do I need to make the current CodeFixProvider abstract and add 2 subclasses?